Frontend/UserTech/src/app/services/bike.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { LoginService } from './login.service';
+import {environment as env} from '../../environments/environment';
+import { HttpClient } from '@angular/common/http';
+import { BikesDTO } from '../dto/bikes-dto';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BikeService {
+  private baseUrl = `${env.apiUrl}/bikes`;
+  
+  constructor(private loginService: LoginService,
+              private http: HttpClient) { }
+
+  getRentedBikes(): Observable<BikesDTO>{
+    if (!this.loginService.isLoggedIn()) {
+      return of();
+    }
+    return this.http.get<BikesDTO>(`${this.baseUrl}/rented`);
```

Comment:
Czy chcemy jawnie sprawdzać czy użytkownik jest zalogowany? Bo samo wywołanie tej metody dla niezalogowanego użytkownika jest czymś błędnym. A jeżeli już pójdzie taki request to incetreptor przechwytuje i reroutuje dla 401.